File: Tetris.Net/[C#]-Tetris.Net/C#/tetris cs/ShapePreview.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace tetris_cs
{
    /// <summary>
    /// Used for displaying a shape preview
    /// </summary>
    class ShapePreview : DataGridView
    {
        //constants used for ignoring DGV focussing
        const int WM_LBUTTONDOWN = 0x201;
        const int WM_LBUTTONDBLCLK = 0x203;
        const int WM_KEYDOWN = 0x100;

        //avoids focussing
        protected override void OnRowPrePaint(DataGridViewRowPrePaintEventArgs e) {
            int p = (int)e.PaintParts;
            p -= (int)DataGridViewPaintParts.Focus;
            e.PaintParts = (DataGridViewPaintParts)p;
		    base.OnRowPrePaint(e);
        }

        //ignores focussing
        protected override void WndProc(ref System.Windows.Forms.Message m)
        {
            if (m.Msg == WM_LBUTTONDBLCLK || m.Msg == WM_LBUTTONDOWN || m.Msg == WM_KEYDOWN)
            {
                return;
            }
            base.WndProc(ref m);
        }


}
}

File: Tetris.Net/[VB]-Tetris.Net/VB/tetris/gui/ShapePreview.vb
''' <summary>
''' Used for displaying a shape preview
''' </summary>
Public Class ShapePreview
    Inherits DataGridView

    'constants used for ignoring DGV focussing
    Const WM_LBUTTONDOWN As Integer = &H201
    Const WM_LBUTTONDBLCLK As Integer = &H203
    Const WM_KEYDOWN As Integer = &H100

    'avoids focussing
    Protected Overrides Sub OnRowPrePaint(ByVal e As System.Windows.Forms.DataGridViewRowPrePaintEventArgs)
        e.PaintParts = e.PaintParts And Not DataGridViewPaintParts.Focus
        MyBase.OnRowPrePaint(e)
    End Sub

    'ignores focussing
    Protected Overrides Sub WndProc(ByRef m As System.Windows.Forms.Message)
        If m.Msg = WM_LBUTTONDBLCLK OrElse m.Msg = WM_LBUTTONDOWN OrElse m.Msg = WM_KEYDOWN Then
            Return
        End If
        MyBase.WndProc(m)
    End Sub

End Class
